src/main/java/com/twilio/oai/TwilioTerraformGenerator.java

```diff
@@ -156,24 +87,58 @@ public void processOpenAPI(final OpenAPI openAPI) {
             }
         }
 
+        // We only need to create resources with full CRUD capabilities.
         removeNonCrudResources(resources);
 
-        resources.values().forEach(resource -> {
+        for (final Iterator<Map.Entry<String, Map<String, Object>>> i = resources.entrySet().iterator();
+            i.hasNext(); ) {
+            final Map<String, Object> resource = i.next().getValue();
+
             final CodegenOperation createOperation = getCodegenOperation(resource, ResourceOperation.CREATE);
             final CodegenOperation updateOperation = getCodegenOperation(resource, ResourceOperation.UPDATE);
 
             // Use the parameters for creating the resource as the resource schema.
             resource.put("schema", createOperation.allParams);
-            for (final CodegenResponse resp : createOperation.responses) {
-                if (resp.is2xx) {
-                    final ArrayList<CodegenProperty> properties = getResponseProperties((Schema<?>) resp.schema,
-                                                                                        getParamNames(createOperation.allParams),
-                                                                                        getParamNames(updateOperation.formParams));
-                    resource.put("responseSchema", properties);
-                    break;
-                }
-            }
-        });
+            createOperation.responses
+                .stream()
+                .filter(r -> r.is2xx)
+                .map(r -> r.schema)
+                .map(Schema.class::cast)
+                .findFirst()
+                .ifPresent(schema -> {
+                    final Map<String, Schema<?>> properties = ModelUtils
+                        .getReferencedSchema(this.openAPI, schema)
+                        .getProperties();
+
+                    final List<CodegenProperty> responseProperties = getResponseProperties(properties,
+                                                                                           getParamNames(createOperation.allParams),
+                                                                                           getParamNames(updateOperation.formParams));
+                    resource.put("responseSchema", responseProperties);
+
+                    // We need to find the parameter to be used as the Terraform resource ID (as it's not always the
+                    // 'sid'). We assume it's the last path parameter for the fetch/update/delete operation.
+                    final CodegenParameter idParameter = updateOperation.pathParams.get(
+                        updateOperation.pathParams.size() - 1);
+                    final String idParameterSnakeCase = toSnakeCase(idParameter.baseName);
```

Comment:
Should probably use `paramName` instead of `baseName`

Reply:
Updated!